### ISSUE TITLE
Rule to enforce consistent spacing after if statement parens when no braces are present

### ISSUE THREAD

**Issue by AbdelrahmanHafez** [eslint/eslint#13119]
**Please describe what the rule should do:**
Enforce consistent spacing after if statement parens when no braces are present.

**What category of rule is this?**
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
```js
if (something)doStuff(); // "warns"
if (something) doStuff(); // "correct usage"
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because it's not specific to a framework/runtime, it's a general JS styling preference.

**Are you willing to submit a pull request to implement this rule?**
I haven't contributed in ESLint before, and quite unfamiliar with the codebase.
I don't mind trying, though.

**Comment by ljharb** [comment]
Would this also apply to `if (something){ doStuff(); }` and enforce `if (something) { doStuff(); }`?

What about `for (;;) doStuff();`, `while (condition) doStuff();`?

**Comment by mdjermanovic** [comment]
> Would this also apply to `if (something){ doStuff(); }` and enforce `if (something) { doStuff(); }`?

This is already covered by [space-before-blocks](https://eslint.org/docs/rules/space-before-blocks):

```js
/*eslint space-before-blocks: "error"*/

if (something){ doStuff(); } // Missing space before opening brace. (space-before-blocks)
```

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgc3BhY2UtYmVmb3JlLWJsb2NrczogXCJlcnJvclwiKi9cblxuaWYgKHNvbWV0aGluZyl7IGRvU3R1ZmYoKTsgfVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by ljharb** [comment]
In that case, this is meant to be a "space-before-implicit-blocks" rule? Perhaps it would make sense to just add an option to `space-before-blocks` to have it apply to implicit blocks also?

**Comment by mdjermanovic** [comment]
Either a new rule or a new option in `space-before-blocks` makes sense to me :+1: 

It also makes sense to target `for`, `while` etc.  with the same rule/option.

If accepted, we should probably check other rules and make sure that this doesn't introduce possible conflicts, e.g., with [keyword-spacing](https://eslint.org/docs/rules/keyword-spacing):

```js
/*eslint keyword-spacing: ["error"]*/

if (something)try { foo() } catch {} // Expected space(s) before "try". (keyword-spacing)
```

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQga2V5d29yZC1zcGFjaW5nOiBbXCJlcnJvclwiXSovXG5cbmlmIChzb21ldGhpbmcpdHJ5IHsgZm9vKCkgfSBjYXRjaCB7fVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMCwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by kaicataldo** [comment]
I would be in favor of updating `space-before-blocks` rather than creating a separate rule.

**Comment by anikethsaha** [comment]
Any updates on this ? 
if no one is taking, I would like to implement this !

CC @kaicataldo @mdjermanovic

**Comment by mdjermanovic** [comment]
@anikethsaha this rule change isn't accepted yet, let's see if it will reach the [consensus](https://eslint.org/docs/maintainer-guide/issues#championing-issues) first.

**Comment by anikethsaha** [comment]
can we open this one ?
